examples/3ds/create_charge.php

```diff
@@ -48,12 +49,30 @@
     null,
     PaymentThreeDS::withThreeDS(
         "https://ec-site.example.com/3ds/complete", // redirect endpoint when 3DS is completed
-        ThreeDSMode::NORMAL() // check documentation for more about 3DS modes
+        ThreeDSMode::NORMAL() // for more details, refer to the Univapay documentation on 3DS modes.
     )
-)->awaitResult(5);
-// Fetch information for issuer token for 3DS authentication and redirect user to 3DS authentication page
-// after 3DS authentication is completed, user will be redirected to the endpoint specified in PaymentThreeDS
-$charge->threeDSIssuerToken();
+);
+
+// This is just an example of how to process with the charge status
+while (1) {
+    $charge = $charge->awaitResult(5); // wait for the charge status to be updated, retrying up to 5 times
+    switch ($charge->status) {
+        case ChargeStatus::PENDING():
+            // still on progress ...
+            break;
+        case ChargeStatus::AWAITING():
+            // Fetch information for issuer token for 3DS authentication and redirect user to 3DS authentication page
+            // after 3DS authentication is completed, user will be redirected to the endpoint specified in PaymentThreeDS
+            $charge->threeDSIssuerToken();
+            break;
+        case ChargeStatus::SUCCESSFUL():
+        case ChargeStatus::FAILED():
+        case ChargeStatus::ERROR():
+        default:
+            goto end;
+    }
+}
+end:
```

Comment:
Not sure that we really need a while loop there to make things more complex... I think something simple like

```php
    $charge = $charge->awaitResult(5);

    switch ($charge->status) {
        case ChargeStatus::AWAITING():
            // Fetch information for issuer token for 3DS authentication and redirect user to 3DS authentication page
            // after 3DS authentication is completed, user will be redirected to the endpoint specified in PaymentThreeDS
            $charge->threeDSIssuerToken();
            break;
        case ChargeStatus::SUCCESSFUL():
            // continue with payment flow
        case ChargeStatus::PENDING():
        case ChargeStatus::FAILED():
        case ChargeStatus::ERROR():
            // implement error handling
    }
```

Reply:
Same with the other examples too.. You'd be surprised how many developers will copy paste your example and just use it as is without any kind of modifications. We don't want to promote bad coding habits or cause their code to be stuck in an endless loop.

Reply:
indeed! i was worried when implementing this on my client side it might fails to catch an updated status, but yes i should make it simple